lib/history/event.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package history
+
+import (
+	"sync"
+	"time"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+)
+
+// Event represents a single timeline event. An event exposes a type and
+// metadata.
+type Event struct {
+	// timeStamp specifies the when the event occurred.
+	timeStamp time.Time
+	// eventType specifies the type of event.
+	eventType EventType
+	// mu locks access to metadata
+	mu sync.Mutex
```

Comment:
I don't think the events need a mutex - the metadata can be passed to an event during construction.